xdsl/transforms/eqsat_serialize_egraph.py

```diff
@@ -0,0 +1,69 @@
+import json
+
+from xdsl.context import Context
+from xdsl.dialects import builtin, eqsat, func
+from xdsl.ir import BlockArgument, Operation
+from xdsl.passes import ModulePass
+
+
+def serialize_func_to_egraph(f_op: func.FuncOp):
+    assert len(f_op.regions) == 1
+    body = f_op.regions[0]
+    assert body.first_block == body.last_block, (
+        "Only single block functions are supported"
+    )
+    body = body.first_block
+
+    i_eclass = 0
+    i_enode = 0
+    eclass_to_id: dict[eqsat.EClassOp, int] = {}
+    op_to_id = {}
+    nodes: list[dict[str, str | list[str]]] = []
+
+    if body:
+        for op in body.ops:
+            if isinstance(op, eqsat.EClassOp):
+                eclass_to_id[op] = (
+                    i_enode  # pick the first e-node as the representative
+                )
+                assert op.operands, "EClassOp must have operands"
```

Comment:
Just a quick question: isn't this enforced by the verifier during loading/IR construction?